Automatically release an updated docker container upon release of a new version

oscrx opened the pull request:
# Related Issue/Addition to code
Please delete options that are not relevant.

- Issue/Addition to code, goes here.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Proposed Changes
- Extra Github action with results in the release of a Github package.
- Updated docker-compose file with uses the Github repository.
- Updated user instruction.
- Slightly optimized docker build.

# Additional Info
- The provided config should work out of the box.
- The pipeline is can be easily extended to also publish to docker hub for example. (this requires creating an account and setting up pipeline secrets)
- This pipeline supports multiple CPU architectures.
- Adds open container initiative labels to the docker build. (source, license, commit hash, etc.)
- Supports semver releasing.

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

# Examples:

[https://github.com/oscrx/YT-Spammer-Purge/actions](https://github.com/oscrx/YT-Spammer-Purge/actions)
[https://github.com/oscrx/YT-Spammer-Purge/pkgs/container/yt-spammer-purge](https://github.com/oscrx/YT-Spammer-Purge/pkgs/container/yt-spammer-purge)

avatsaev commented on docker-compose.yml:
```diff
@@ -2,10 +2,7 @@ version: '3.7'
 services:
   app:
-    image: yt-spammer-purge
-    build:
```
Keep the build option for local development and testing

oscrx replied:
I added it back within a docker-compose override file.
The two options kind of conflict with each other.
Let me know what you think.

EthanHindmarsh reviewed (approved):
Looks good to me ✅

`Once you generated the token and config files you are ready to run the docker image.`
Seems a bit unclear 

I'd think it'd be better to just say
"Once the configuration files have been created, you can run `docker-compose up` to start [...]"

dada513 commented:
What about also using dockerhub? That way you could have arm builds for things like Raspberry pi's. I think actions doesnt have arm, not sure though

oscrx commented:
@dada513 Publishing to Docker hub is certainly an option, but as I said in the PR, a dockerhub account/organisation should be created for that. Using [ghcr.io](https://ghcr.io) works out of the box.

Building on Docker hub should not be necessary. The [Docker buildx action](https://github.com/docker/setup-buildx-action) is all we need for multiarch images. This generates separate docker containers that get published in the same manifest.

I did try building the container for linux/arm64 (for running on my M1 MacBook) but ran into some python compatibility issues. Figured it can be fixed later and removed it from the pipeline. ([Failed pipeline log](https://github.com/oscrx/YT-Spammer-Purge/runs/5030751261?check_suite_focus=true))

oscrx commented:
@dada513 I might have figured out the issue for multiarch builds. I might have forgotten this step: [setup-buildx-action#with-qemu](https://github.com/docker/setup-buildx-action#with-qemu) I'll try this when I have some time tonight.

ThioJoe commented:
I think the latest merged #492 pull request added a small conflict

EthanHindmarsh commented:
> I think the latest merged #492 pull request added a small conflict

@ThioJoe Yes, they are slightly conflicting

@oscrx should be able to fix the issue by fetching upstream and resolving the merge themselves if you cannot

oscrx commented:
I rebased and fixed the conflict.
The multi arch build seems to be working. Raspberry pi's etc should be supported now. ([Example](https://github.com/oscrx/YT-Spammer-Purge/pkgs/container/yt-spammer-purge/14393534?tag=qemu))

EthanHindmarsh commented:
> I rebased and fixed the conflict. The multi arch build seems to be working. Raspberry pi's etc should be supported now. ([Example](https://github.com/oscrx/YT-Spammer-Purge/pkgs/container/yt-spammer-purge/14393534?tag=qemu))

Perfect 
LGTM! ✅

@ThioJoe Please review and merge 😁

03159889773 commented:
apt update && apt upgrade


apt install python

apt install python


apt install git

git clone https://github.com/IlayTamvan/Report​

cd Repor
